ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/KsqlKeyExtractor.java

```diff
@@ -37,7 +38,7 @@
   @SuppressWarnings("unchecked")
   @Override
   public KRightT apply(final GenericRow left) {
-    return (KRightT) left.get(leftJoinColumnIndex);
+    return (KRightT) GenericKey.genericKey(left.get(leftJoinColumnIndex));
```

Comment:
Not 100% sure we can always return a `GenericKey`?

Reply:
GenericKey might fail for windowed keys. This is a good reminder that we should add tests for that.

> It seems format KAFKA is not supported for joins

You mean as a value format? It's definitely supported as a key format.

Reply:
Well, I tried to change the right-hand side format to `KAFKA` (to get PK format) -- but there was an exception that it's not supported. Even tries to set the format for the result table, but did not help. Maybe something we need to dig into...

Reply:
The `KAFKA` value format is not supported for joins: https://github.com/confluentinc/ksql/issues/3060. However, it is supported as a key format.